pkg/cluster/storageaccounts.go

```diff
@@ -70,10 +70,18 @@ func (m *manager) enableServiceEndpoints(ctx context.Context) error {
 	return nil
 }
 
+func (m *manager) isWorkerProfileAvailable() bool {
+	return len(m.doc.OpenShiftCluster.Properties.WorkerProfiles) == 0
+}
+
 // migrateStorageAccounts redeploys storage accounts with firewall rules preventing external access
 // The encryption flag is set to false/disabled for legacy storage accounts.
 func (m *manager) migrateStorageAccounts(ctx context.Context) error {
 	resourceGroup := stringutils.LastTokenByte(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
+	if m.isWorkerProfileAvailable() {
+		m.log.Error("Unable to create storage resource due to missing worker profiles.")
```

Comment:
We should be stating that we are `Skipping migrateStorageAccounts due to missing worker profiles` or similar wording.

Reply:
@darthhexx : done, could you please verify the changes?

Reply:
Changes look great. We can merge once CI goes green.